pallets/cash/src/internal/balance_helpers.rs

```diff
@@ -0,0 +1,535 @@
+use crate::{
+    reason::{MathError, Reason},
+    types::{
+        AssetAmount, AssetBalance, AssetQuantity, CashPrincipal, CashPrincipalAmount, Quantity,
+    },
+};
+use our_std::{cmp::min, convert::TryFrom};
+
+// XXX use Balances instead of raw balances everywhere and put all fns on types?
```

Comment:
I removed this in my branch, I think we're better off with these functions as they make the intentions very clear - I started to convert it to using impl fns on the types and it became a lot less obvious